File: Exemplos M10/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Kotlin-font/app/src/main/java/com/elgin/intent_digital_hub/Activities/SAT/SatModel.kt
package com.elgin.intent_digital_hub.Activities.SAT


/**
 * Os dois modelos diferentes de SAT possuem xmls diferentes no envio de venda, o enumerator facilita guardando também o nome dos arquivos xml de envio de envio de venda para cada modelo, encontrados em res/raw/
 */
enum class SatModel(val SALE_XML_ARCHIVE_NAME: String) {
    SMART_SAT("sat_enviar_dados_venda"),
    SAT_GO("satgo_enviar_dados_venda");


    open fun SALE_XML_ARCHIVE_NAME(): String? {
        return SALE_XML_ARCHIVE_NAME
    }

}

File: Exemplos M10/App_IntentDigitalHUB_XamarinAndroid/Elgin_IntentDigitalHUB_XamarinAndroid-font/Sat/SatModel.cs
namespace Xamarin_Android_Intent_Digital_Hub.Sat
{
    /**
     * Os dois modelos diferentes de SAT possuem xmls diferentes no envio de venda, o enumerator facilita guardando também o nome dos arquivos xml de envio de envio de venda para cada modelo, encontrados em res/raw/
     */
    class SatModel
    {
        private SatModel(string SALE_XML_ARCHIVE_NAME) { this.SALE_XML_ARCHIVE_NAME = SALE_XML_ARCHIVE_NAME; }

        public string SALE_XML_ARCHIVE_NAME { get; private set; }

        public static SatModel SMART_SAT = new SatModel("sat_enviar_dados_venda");
        public static SatModel SAT_GO = new SatModel("satgo_enviar_dados_venda");
    }
}